### ISSUE TITLE
ULID from Timestamp and String?

### ISSUE THREAD

**Issue by aadrian** [f4b6a3/ulid-creator#25]
Hi,

Is there any way to create an ULID from a Timestamp **and** a String (that gets "hashed", to fit as part of the ULID) ?

e.g.
```java
UlidCreator.getUlid(long time,  String originalId)
```
So that when calling this would return always the same ULID?

Thank you.

**Comment by fabiolimace** [comment]
Hi @aadrian 

You can implement your own ULID generator like this method:

```java
    public static Ulid generate(final long time, final String string) {

        // Generate a SHA-256 hash of the string
        MessageDigest hasher = hasher("SHA-256");
        hasher.update(string.getBytes(StandardCharsets.UTF_8));
        ByteBuffer hash = ByteBuffer.wrap(hasher.digest());

        // Concatenate the time and the first 10 bytes of the hash
        final long msb = (time << 16) | (hash.getShort() & 0xffffL);
        final long lsb = hash.getLong();

        return new Ulid(msb, lsb);
    }
```

Usage example:

```java
    long time = file.getCreatedAt();
    String name = file.getFileName();
    Ulid ulid = HashUlid.generate(time, name);
```

A complete example can be found here: https://gist.github.com/fabiolimace/3e2d3363720c4cd776c2b0762c94fda8.

I don't think it should be added to the library as it is a specific use case.

Let me know if this suggestion was helpful.

Best regards

**Comment by fabiolimace** [comment]
You changed my mind. I will add Hash-based ULID in the next library version soon. I just have to write unit tests and update the documentation as the method is already done.

Thanks!

**Comment by fabiolimace** [comment]
New static methods:

```java
Ulid ulid = UlidCreator.getHashUlid(time, string);
```

```java
Ulid ulid = UlidCreator.getHashUlid(time, bytes);
```

**Comment by fabiolimace** [comment]
Released v5.2.0! 🎉
